src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/Passes/RazorFormattingPass.cs

```diff
@@ -403,6 +403,13 @@ private bool FormatBlock(FormattingContext context, RazorSourceDocument source,
             {
                 var openBraceLineNumber = openBraceNode.GetLinePositionSpan(source).Start.Line;
                 var openBraceLine = source.Text.Lines[openBraceLineNumber];
+                // The open brace node might actually start with a newline on the line before, which could be blank,
+                // so make sure we find some actual content.
+                while (!openBraceLine.GetFirstNonWhitespacePosition().HasValue)
+                {
+                    openBraceLine = source.Text.Lines[++openBraceLineNumber];
+                }
+
                 Debug.Assert(openBraceLine.GetFirstNonWhitespacePosition().HasValue);
                 additionalIndentation = source.Text.GetSubTextString(TextSpan.FromBounds(openBraceLine.Start, openBraceLine.GetFirstNonWhitespacePosition().GetValueOrDefault()));
```

Comment:
Is it possible avoid calling `GetFirstNonWhitespacePosition()` multiple times for the same line? It seems like this could be tightened up a bit.

Reply:
Switched to using (your improved) GetLeadingWhitespace method